### ISSUE TITLE
radar graphics glitch when labels turned on

### ISSUE THREAD

**Issue by chpoklv** [openscope/openscope#814]
Browser: google chrome Version 61.0.3163.100 (Official Build) (32-bit)

Airport: any

Approximate play time before issue: i was playing for few days ~one year ago

#### Issue Description
Some airports crashes radar graphics when i turn on fix and runway labels. Marks appear not matching with terrain, and things get worse when i try to move radar with right mouse button- everything starts glitching including rotating lines and labels, sometimes it just freezes. The only thing that keeps moving adequately in any case is airplane marks. I found that happening on particular airports, i have checked some of them:

Dublin- OK
shannon- OK
Seattle- OK

luisville- crashes
albuquerque - crashes
El Paso- crashes
Portland- crashes
Tampa-crashes
Tucson-crashes
Ted stevens- crashes
Boston-crashes

If i open/refresh page on normal airport->works fine
if i turn on labels on normal airport->works fine
if i switch to glitchy airport after normal airport->works fine without labels
if i turn on labels on glitchy airport->radar crashes
if i turn off labels on glitchy airport->it keeps glitching
if i switch to normal airport with crashed radar graphics->keeps glitching
if i switch to normal airport, then refresh page->works fine again

![radar graph crashes](https://user-images.githubusercontent.com/33282672/32272651-af5482ba-bf07-11e7-86ed-bc1d0c49ed2a.jpg)
.

**Comment by n8rzz** [comment]
Thanks for the detailed bug report!  We just deployed `v5.6.0` last night, so there may be some bugs (though we fixed the ones we saw in testing).

Two additional questions for you: 
1) are there any console errors when you see the glitchy drawings? _(ctrl + alt + i or cmd + alt + i) to open dev tools, then navigate to console tab_
1) are any of the airports experiencing this issue labeled as WIP airports?

**Comment by Fechulo** [comment]
I'll answer those questions.
Surprisingly, no console errors and it occurs with the vast majority of airports, wip and non wip.
I'm surprised we didn't notice this in testing

**Comment by n8rzz** [comment]
@Fechulo @chpoklv humor me a moment, but can you both try again after clearing your cache?  shift + f5 or cmd + shift + r on mac.

I didn't see any issues with albuquerque or tuscon after some quick pan and zoom tests

**Comment by chpoklv** [comment]
have tried shift+f5 and chrome incognito window, same problem. found this glitch even on android chrome
![screenshot_2017-11-01-16-07-05](https://user-images.githubusercontent.com/33282672/32278705-2e166818-bf1f-11e7-868b-5e8ceea0d089.png)

**Comment by n8rzz** [comment]
Hmm 🤔 

Based on those screenshots it looks like some funkiness with translations.  In this latest version, where those happen has moved and is designed to be much more modular.  A stale cache _seems_ like the issue here.  Especially since I'm not seeing it at all on my end in `prod` or `staging`.

But, from what I'm reading here, you are not seeing the same results.  that makes me 😞 

@Fechulo are you able to reproduce this as well?

**Comment by n8rzz** [comment]
I can repro this locally now.  This is a *_major_* issue that we will be looking into immediately and will be pushed through as quick as possible.

**Comment by n8rzz** [comment]
@chpoklv we have identified the issue and I have opened a PR for review.  Since you were the original reporter, could you take a look at the review app for this PR?  

https://openscope-dev-pr-815.herokuapp.com/

This version should have the issue you reported fixed.  If you still see the issue there, report back here and let us know.

**Comment by chpoklv** [comment]
I see this issue fixed now.

I notice another small bug with graphics: heading scale around the radar screen is drawn only after you move a screen with right mouse button, i guess it should appear when you select the aircraft or maybe it should be visible all the time?

**Comment by n8rzz** [comment]
@chpoklv could you open a new issue for that one?  I'm pretty sure I know why that's happening, but I would consider it a low priority (for me) at the moment.

Either way, open a new issue for it and I'll add some notes so anyone could go in and fix it.

**Comment by erikquinn** [comment]
Resolved by #815. Closing.
